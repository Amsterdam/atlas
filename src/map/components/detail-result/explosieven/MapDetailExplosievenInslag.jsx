import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from '../MapDetailResultItem';

const MapDetailExplosievenInslag = ({ panoUrl, inslag }) => (
  <section className="map-detail-result">
    <header
      className={`
        map-detail-result__header
        map-detail-result__header--${panoUrl ? 'pano' : 'no-pano'}
      `}
    >
      {panoUrl && (
        <img
          alt="Panoramabeeld"
          className="map-detail-result__header-pano"
          src={panoUrl}
        />
      )}
      <div className="map-detail-result__header-container">
        <h1 className="map-detail-result__header-title">Inslag</h1>
        <h2 className="map-detail-result__header-subtitle">{inslag.label}</h2>
      </div>
    </header>
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Datum van inslag"
        value={inslag.dateInslag}
      />
      <MapDetailResultItem
        label="Soort handeling"
        value={inslag.type}
      />
      <MapDetailResultItem
        label="Bron"
        value={inslag.source}
      />
    </ul>
  </section>
);

MapDetailExplosievenInslag.defaultProps = {
  panoUrl: ''
};

MapDetailExplosievenInslag.propTypes = {
  inslag: PropTypes.shape({
    dateInslag: PropTypes.string,
    label: PropTypes.string,
    source: PropTypes.string,
    type: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string
};

export default MapDetailExplosievenInslag;
