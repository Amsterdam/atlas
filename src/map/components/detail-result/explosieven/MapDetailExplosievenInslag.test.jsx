import React from 'react';
import { shallow } from 'enzyme';

import MapDetailExplosievenInslag from './MapDetailExplosievenInslag';

describe('MapDetailExplosievenInslag', () => {
  describe('rendering', () => {
    it('should render everything', () => {
      const inslag = {
        date: new Date('1980-08-19'),
        label: 'Inslag label',
        remarks: 'Inslag remarks',
        source: 'Inslag source',
        type: 'Inslag type'
      };
      const panoUrl = 'panoUrl';
      const wrapper = shallow(
        <MapDetailExplosievenInslag
          panoUrl={panoUrl}
          inslag={inslag}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should render without pano url', () => {
      const inslag = {
        date: new Date('1980-08-19'),
        label: 'Inslag label',
        remarks: 'Inslag remarks',
        source: 'Inslag source',
        type: 'Inslag type'
      };
      const wrapper = shallow(
        <MapDetailExplosievenInslag
          inslag={inslag}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
