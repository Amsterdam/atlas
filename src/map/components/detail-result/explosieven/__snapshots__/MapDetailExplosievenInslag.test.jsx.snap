// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailExplosievenInslag rendering should render everything 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--pano
      "
  >
    <img
      alt="Panoramabeeld"
      className="map-detail-result__header-pano"
      src="panoUrl"
    />
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        Inslag
      </h1>
      <h2
        className="map-detail-result__header-subtitle"
      >
        Inslag label
      </h2>
    </div>
  </header>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultDateItem
      date={1980-08-19T00:00:00.000Z}
      label="Datum van inslag"
    />
    <MapDetailResultItem
      label="Soort handeling"
      value="Inslag type"
    />
    <MapDetailResultItem
      label="Bron"
      value="Inslag source"
    />
    <MapDetailResultItem
      label="Opmerkingen"
      value="Inslag remarks"
    />
  </ul>
</section>
`;

exports[`MapDetailExplosievenInslag rendering should render without pano url 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--no-pano
      "
  >
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        Inslag
      </h1>
      <h2
        className="map-detail-result__header-subtitle"
      >
        Inslag label
      </h2>
    </div>
  </header>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultDateItem
      date={1980-08-19T00:00:00.000Z}
      label="Datum van inslag"
    />
    <MapDetailResultItem
      label="Soort handeling"
      value="Inslag type"
    />
    <MapDetailResultItem
      label="Bron"
      value="Inslag source"
    />
    <MapDetailResultItem
      label="Opmerkingen"
      value="Inslag remarks"
    />
  </ul>
</section>
`;
